File: hospitalManagement/Faculties.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Faculties : IBasicActivities<Faculty>
    {
        //Field	
        private List<Faculty> facultyList;
        private long count;



        // Infor field	
        //  Management field	
        // Dynamic field	
        // Properties	
        public long Count { get => count; set => count = value; }
        internal List<Faculty> FacultyList { get => facultyList; set => facultyList = value; }
        // Constructors	

        public Faculties()
        {
            facultyList = new List<Faculty>();
            this.Count = 0;
        }
        public Faculties(List<Faculty> facultyList, long count)
        {
            this.FacultyList = facultyList;
            this.Count = count;
        }

        // Destructors	
        ~Faculties() { }
        // Methods	
        // in, output	
        // General method	
        // Other method	
        public void AddItem()
        {
            Faculty faculty = new Faculty();
            Console.WriteLine("New faculty");
            faculty.Input();
            facultyList.Add(faculty);
            Console.WriteLine("Done!");
        }

        public void AddItem(Faculty value)
        {
            Console.WriteLine("New faculty");
            facultyList.Add(value);
            Console.WriteLine("Done!");
        }

        public bool Clear()
        {
            Console.WriteLine("Clear all faculties");

            facultyList.Clear();
            if (facultyList.Count > 0)
            {
                Console.WriteLine("Failure!");
                return false;
            }
            this.Count = 0;
            Console.WriteLine("Done!");

            return true;
        }

        public Faculty FindItem(string id)
        {
            Console.WriteLine("Find the faculty");

            Faculty res = null;
            facultyList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found faculty with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");
            }
            return res;
        }

        public bool RemoveItem(string id)
        {
            Console.WriteLine("Remove the faculty");

            bool res = false;
            facultyList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    facultyList.Remove(value);
                    res = true;
                }
            });
            if (res == false)
            {
                Console.WriteLine($"Not found faculty with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public void ShowInformation()
        {
            Console.WriteLine("-------------------------------------------------------------------");

            Console.WriteLine("Show information of all faculties");

            facultyList.ForEach(value =>
            {
                value.Output();
                Console.WriteLine();
            });
            Console.WriteLine("Done!");
            Console.WriteLine("-------------------------------------------------------------------");
        }

        public Faculty UpdateItem(string id)
        {
            Console.WriteLine("Update the faculty");

            Faculty res = null;
            facultyList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    value.Input();
                    res = value;
                }

            });
            if (res == null)
            {
                Console.WriteLine($"Not found the faculty with id: {id}");
            }
            else
            {
                Console.WriteLine("Done!");
            }
            return res;
        }
        // Overriding	

    }
}

File: src/hospitalManagement/Faculties.java
package hospitalManagement;


import java.util.ArrayList;
class Faculties 
    {
        //Field	
        private ArrayList<Faculty> facultyList;
        private long count;



        // Information field	
        //  Management field	
        // Dynamic field	
        // Properties	
        public ArrayList<Faculty> getFacultyList() {
            return facultyList;
        }
        public void setFacultyList(ArrayList<Faculty> facultyList) {
            this.facultyList = facultyList;
        }
        public long getCount() {
            return count;
        }
        public void setCount(long count) {
            this.count = count;
        }
        // Constructors	

        public Faculties ()
        {
            facultyList = new ArrayList<Faculty>();
            count = 0;
        }
        
        public Faculties(ArrayList<Faculty> facultyList, long count)
        {
            this.facultyList = facultyList;
            this.count = count;
        }

        // Methods	
        // in, output	
        // General method	
        // Other method	
        public void AddItem()
        {
            Faculty faculty = new Faculty();
            System.out.println("New faculty");
            faculty.Input();
            facultyList.add(faculty);
            System.out.println("Done!");
        }

        public void AddItem(Faculty value)
        {
            System.out.println("New faculty");
            facultyList.add(value);
            System.out.println("Done!");
        }

        public Boolean Clear()
        {
            System.out.println("Clear all faculties");

            facultyList.clear();
            if (facultyList.size() > 0)
            {
                System.out.println("Failure!");
                return false;
            }
            this.count = 0;
            System.out.println("Done!");

            return true;
        }

        public Faculty FindItem(String id)
        {
            System.out.println("Find the faculty");

            Faculty res = null;
            
            for (Faculty i : facultyList) {
                if( i.getId() == id)
                {
                    res = i;
                }
            }
            if (res == null)
            {
                System.out.println("Not found faculty");

            }
            else
            {
                System.out.println("Done!");
            }
            return res;
        }

        public boolean RemoveItem(String id)
        {
            System.out.println("Remove the faculty");

            boolean res = false;
            for (Faculty i : facultyList) {
                if( i.getId() == id)
                {
                    facultyList.remove(i);
                    this.count--;
                }
            }
            if (res == false)
            {
                System.out.println("Not found faculty");

            }
            else
            {
                System.out.println("Done!");

            }
            return res;
        }

        public void ShowInformation()
        {
            System.out.println("-------------------------------------------------------------------");

            System.out.println("Show information of all faculties");

            facultyList.forEach(value -> { value.Output(); System.out.println(); });

            System.out.println("Done!");
            System.out.println("-------------------------------------------------------------------");
        }

        public Faculty UpdateItem(String id)
        {
            System.out.println("Update the faculty");

            Faculty res = null;
            for (Faculty i : facultyList) {
                if( i.getId() == id)
                {
                    i.Input();
                    res = i;
                }
            }
            if (res == null)
            {
                System.out.println("Not found the faculty with id: {id}");
            }
            else
            {
                System.out.println("Done!");
            }
            return res;
        }
        // Overriding	

    }